src/test/java/com/google/devtools/build/lib/remote/RemoteModuleTest.java

```diff
@@ -72,14 +73,22 @@
 import java.net.URI;
 import java.time.Duration;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
 
 /** Tests for {@link RemoteModule}. */
-@RunWith(JUnit4.class)
+@RunWith(Parameterized.class)
```

Comment:
I would rather not change this giant test suite to parameterized. The fact that you have to add many `if`s is the symptom that the test cases become hard to read and maintain. Also, remaining test cases that don't use the test parameters are executed multiple times.

Reply:
With additional branch circuitBreaker number of test increased two folds (if I try to verify with every branch) and file was already too big, so I decided to use parameterized test. Also I had to use `if-else` for different assert statements.
 
I agree with your other point that `testNetrc_netrcWithoutRemoteCache` is unnecessarily executed multiple times, it would have better if junit5 was already integrated.

Updated as per your suggestion.